lib/auth/auth_with_roles.go

```diff
@@ -192,15 +193,32 @@ func (a *ServerWithRoles) actionWithExtendedContext(namespace, kind, verb string
 }
 
 // actionForKindSession is a special checker that grants access to session
-// recordings.  It can allow access to a specific recording based on the
+// recordings. It can allow access to a specific recording based on the
 // `where` section of the user's access rule for kind `session`.
-func (a *ServerWithRoles) actionForKindSession(namespace string, sid session.ID) error {
+func (a *ServerWithRoles) actionForKindSession(namespace string, sid session.ID) (types.SessionKind, error) {
+	sessionEnd, err := a.findSessionEndEvent(namespace, sid)
+
 	extendContext := func(ctx *services.Context) error {
```

Comment:
what happens when `a.findSessionEndEvent` fails? in rare case an empty session kind can be logged?

Reply:
If it fails, the [`actionWithExtendedContext` will return an error](https://github.com/gravitational/teleport/blob/56d85facba2661811dc69f576056a253a04fd331/lib/auth/auth_with_roles.go#L189), and we don't emit an audit event [if that returns an error](https://github.com/gravitational/teleport/blob/56d85facba2661811dc69f576056a253a04fd331/lib/auth/auth_with_roles.go#L5931-L5935). So, in the end, we won't even have the audit event.